### PR TITLE
Audit filters specifics on case report

### LINKED ISSUE #4219: Audit clinical filter on variantS page
Eg add clinical filter to the dropdown, or possibly add a small checkbox/extra button. 

Screenshot 2023-11-21 at 13 06 22

It is also very relevant to know what variantS type the filter was audited from in general (SNV, STR, SV...), so perhaps add that automagically on the reports rather than having it in the filter name.

### PR DESCRIPTION
**Pull request by northwestwitch**
Attempt at adding info on variants category and type for audited filters on case report (fix #4219):

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN, CR

### CODE REVIEW

File: scout/server/app.py [hunk 1]
```diff
@@ -32,11 +33,6 @@
     variants,
 )
 
-try:
-    from urllib.parse import unquote
-except ImportError:
-    from urllib2 import unquote
```
Comment (northwestwitch) [inline scout/server/app.py:38]: We no longer support Python 2, no?

File: scout/adapter/mongo/filter.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def audit_filter(self, filter_id, institute_obj, case_obj, user_obj, category="s
             target = VARIANTS_TARGET_FROM_CATEGORY.get(category)
 
             case_name = case_obj.get("display_name")
-            link = url_for(target, case_name=case_name, institute_id=institute_obj.get("_id"))
+            link = url_for(target, case_name=case_name, **filter_obj)
```
Comment (northwestwitch) [inline scout/adapter/mongo/filter.py:141]: I need to change this and collect the info from the variants page I'm on instead!

File: scout/adapter/mongo/filter.py [hunk 3]
```diff
@@ -4,17 +4,9 @@
 from bson.objectid import ObjectId
 from flask import url_for
 
-LOG = logging.getLogger(__name__)
+from scout.constants.variant_tags import VARIANTS_TARGET_FROM_CATEGORY
 
-VARIANTS_TARGET_FROM_CATEGORY = {
```
Comment (northwestwitch) [inline scout/adapter/mongo/filter.py:9]: Moved to constants to be reused

File: scout/adapter/mongo/filter.py [hunk 4]
```diff
@@ -4,17 +4,9 @@
 from bson.objectid import ObjectId
 from flask import url_for
 
-LOG = logging.getLogger(__name__)
+from scout.constants.variant_tags import VARIANTS_TARGET_FROM_CATEGORY
```
Comment (dnil) [inline scout/adapter/mongo/filter.py:7]: We normally import to the main constants, and export them from there, dont we?

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 5]
```diff
@@ -261,9 +261,11 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
                 </tr>
               </thead>
               <tbody>
+                 {% set audit_query = namespace() %}
                  {% for audit in audits %}
+                  {% set audit_query = audit.link|url_args %}
                   <tr>
-                    <td><strong>{{ audit.subject }}</strong> was marked checked by {{ audit.user_name }} on {{audit.created_at.strftime('%Y-%m-%d')}}.</td>
+                    <td><strong>{{ audit.subject }} (category:{{ audit_query.category if audit_query.category else "NA"}}, type:{{audit_query.variant_type if audit_query.variant_type else "NA"}})</strong> was marked checked by {{ audit.user_name }} on {{audit.created_at.strftime('%Y-%m-%d')}}.</td>
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/case_report.html:268]:
```suggestion
                    <td><strong>{{ audit.subject }} ({{audit_query.variant_type if audit_query.variant_type else "type unavailable -"}} {{ audit_query.category if audit_query.category else "category unavailable"}})</strong> was marked checked by {{ audit.user_name }} on {{audit.created_at.strftime('%Y-%m-%d')}}.</td>
```
Lets save some ink - stuff like "clinical SV" is not really made more clear with "category" and "type" added.

File: scout/adapter/mongo/filter.py (hunk 4 above)
Comment (dnil) [inline scout/adapter/mongo/filter.py:7]:
```suggestion
from scout.constants import VARIANTS_TARGET_FROM_CATEGORY
```

File: scout/server/blueprints/variants/controllers.py [hunk 6]
```diff
@@ -27,6 +27,7 @@
     SPIDEX_HUMAN,
     VARIANT_FILTERS,
 )
+from scout.constants.variant_tags import VARIANTS_TARGET_FROM_CATEGORY
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:30]:
```suggestion
from scout.constants.variant_tags import VARIANTS_TARGET_FROM_CATEGORY
from scout.constants import CANCER_EXPORT_HEADER, EXPORT_HEADER
```

Comment (dnil) [reply scout/server/blueprints/variants/controllers.py:30]: Ahem, seems we already forgot that before, but we did do the import to `scout.constants` with `CANCER_EXPORT_HEADER` etc so we might as well.

File: scout/server/app.py [hunk 7]
```diff
@@ -242,6 +238,11 @@ def url_decode(string):
         """Decode a string with encoded hex values."""
         return unquote(string)
 
+    @app.template_filter()
+    def url_args(url: str) -> Dict[str, str]:
```
Comment (dnil) [inline scout/server/app.py:242]: 😸

**Review by dnil — APPROVED** [review]
This works just fine! A couple of suggestions on the report output and the imports. The imports are really only style - it may have been to make refactoring easier way back when, but with modern editors that is easy enough anyway.

File: scout/adapter/mongo/filter.py (hunk 4 above)
Comment (northwestwitch) [reply scout/adapter/mongo/filter.py:7]: Ah OK, I'll change it like that!

**Pull request merged**